src/genotyper.cc

```diff
@@ -737,6 +769,12 @@ Status genotype_site(const genotyper_config& cfg, MetadataCache& cache, BCFData&
         losses.insert(make_pair(sample_name,loss));
     }
     merge_loss_stats(losses, losses_for_site);
+
+    if (residuals != nullptr && any_losses(losses)) {
```

Comment:
(correct me if this is wrong) could we replace the call to `any_losses` with `!residuals->empty()` and eliminate the `any_losses` function?